db/compaction_iterator.cc

```diff
@@ -563,10 +588,13 @@ void CompactionIterator::PrepareOutput() {
   }
 }
 
-inline SequenceNumber CompactionIterator::findEarliestVisibleSnapshot(
+SequenceNumber CompactionIterator::findEarliestVisibleSnapshot(
     SequenceNumber in, SequenceNumber* prev_snapshot) {
   assert(snapshots_->size());
-  SequenceNumber prev __attribute__((__unused__)) = kMaxSequenceNumber;
+  if (snapshot_checker_ != nullptr) {
+    return findEarliestVisibleSnapshotWithSnapshotChecker(in, prev_snapshot);
+  }
+  SequenceNumber prev = kMaxSequenceNumber;
   for (const auto cur : *snapshots_) {
     assert(prev == kMaxSequenceNumber || prev <= cur);
     if (cur >= in) {
```

Comment:
The only difference between this func and findEarliestVisibleSnapshotWithSnapshotChecker is the check in this line. Can we instead of adding findEarliestVisibleSnapshotWithSnapshotChecker, simply put the if-then condition here: i.e., if (snapshot_checker_ != nullptr) call snapsh_checker else (cur >= in).